client/src/main/java/org/spine3/base/MapStringifier.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.base;
+
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static org.spine3.util.Exceptions.newIllegalArgumentException;
+
+/**
+ * The stringifier for the {@code Map} classes.
+ *
+ * <p>The stringifier for the type of the elements in the map
+ * should be registered in the {@code StringifierRegistry} class
+ * for the correct usage of {@code MapStringifier}.
+ *
+ * <h3>Example</h3>
+ *
+ * {@code
+ *  // The registration of the stringifier.
+ *  final Type type = new TypeToken<Map<String, Long>>(){}.getType();
+ *  StringifierRegistry.getInstance().register(stringifier, type);
+ *
+ *  // Obtain already registered `MapStringifier`.
+ *  final Stringifier<Map<String, Long>> mapStringifier = StringifierRegistry.getInstance()
+ *                                                                           .getStringifier(type);
+ *
+ *  // Convert to string. The converted string is result of the {@code Map#toString()} method.
```

Comment:
As discussed vocally, let's try to make the conversion operations (forward->backward) symmetrical, so that `doBackward(doForward(A)) equals A` and `doForward(doBackward(A)) equals A`.